src/test/regress/expected/distributed_types_conflict.out

```diff
@@ -0,0 +1,80 @@
+(2 rows)
+
+-- create a type on a worker that should not cause data loss once overwritten with a type
+-- from the coordinator
+\c - - - :worker_1_port
+SET citus.enable_ddl_propagation TO off;
+SET search_path TO type_conflict;
+CREATE TYPE my_precious_type AS (secret text, should bool);
+CREATE TABLE local_table (a int, b my_precious_type);
+INSERT INTO local_table VALUES (42, ('always bring a towel', true)::my_precious_type);
+\c - - - :master_port
+SET search_path TO type_conflict;
+-- overwrite the type on the worker from the coordinator. The type should be over written
+-- but the data should not have been destroyed
+CREATE TYPE my_precious_type AS (scatterd_secret text);
+-- verify the data is retained
+\c - - - :worker_1_port
+SET search_path TO type_conflict;
+\d+ local_table
+                                          Table "type_conflict.local_table"
+ Column |                Type                | Collation | Nullable | Default | Storage  | Stats target | Description 
+--------+------------------------------------+-----------+----------+---------+----------+--------------+-------------
+ a      | integer                            |           |          |         | plain    |              | 
+ b      | "my_precious_type(citus_backup_0)" |           |          |         | extended |              | 
+
+SELECT * FROM local_table;
+ a  |             b              
+----+----------------------------
+ 42 | ("always bring a towel",t)
+(1 row)
+
+\c - - - :master_port
+SET search_path TO type_conflict;
+-- make sure worker_create_or_replace correctly generates new names while types are existing
+SELECT worker_create_or_replace_object('CREATE TYPE type_conflict.multi_conflicting_type AS (a int, b int);');
+ worker_create_or_replace_object 
+---------------------------------
+ t
+(1 row)
+
+SELECT worker_create_or_replace_object('CREATE TYPE type_conflict.multi_conflicting_type AS (a int, b int, c int);');
+ worker_create_or_replace_object 
+---------------------------------
+ t
+(1 row)
+
+SELECT worker_create_or_replace_object('CREATE TYPE type_conflict.multi_conflicting_type AS (a int, b int, c int, d int);');
+ worker_create_or_replace_object 
+---------------------------------
+ t
+(1 row)
+
+SELECT worker_create_or_replace_object('CREATE TYPE type_conflict.multi_conflicting_type_with_a_really_long_name_that_truncates AS (a int, b int);');
+ worker_create_or_replace_object 
+---------------------------------
+ t
+(1 row)
+
+SELECT worker_create_or_replace_object('CREATE TYPE type_conflict.multi_conflicting_type_with_a_really_long_name_that_truncates AS (a int, b int, c int);');
+ worker_create_or_replace_object 
+---------------------------------
+ t
+(1 row)
+
+SELECT worker_create_or_replace_object('CREATE TYPE type_conflict.multi_conflicting_type_with_a_really_long_name_that_truncates AS (a int, b int, c int, d int);');
+ worker_create_or_replace_object 
+---------------------------------
+ t
+(1 row)
```

Comment:
Minor: should we have something like `SELECT count(*) FROM pg_types WHERE typename ILIKE 'multi_conflicting_type%base%'`  (the query might not be accurate) -- so that we ensure that we create the types.